eclipse-collections/src/main/java/org/eclipse/collections/impl/collection/mutable/CollectionAdapter.java

```diff
@@ -44,7 +44,7 @@
  * {@link #wrapList(Iterable)} factory method. To wrap a collection with the MutableCollection interface alone, use
  * the {@link #adapt(Collection)} factory method.
  */
-public final class CollectionAdapter<T>
+public class CollectionAdapter<T>
         extends AbstractCollectionAdapter<T>
```

Comment:
Note how CollectionAdapter extends AbstractCollectionAdapter. This is the pattern I think we should copy. The superclass is public api.